rust/definitions/src/navigation.rs

```diff
@@ -0,0 +1,659 @@
+}
+
+#[derive(Clone, PartialEq)]
+pub struct LogScreenEntry {
+    pub order: u32,
+    pub timestamp: String,
+    pub events: Vec<Event>,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum ScreenData {
+    Scan,
+    Keys { f: MKeys },
+    Settings { f: MSettings },
+    Log { f: MLog },
+    LogDetails { f: MLogDetails },
+    Transaction { f: MTransaction },
+    SeedSelector { f: MSeeds },
+    KeyDetails { f: MKeyDetails },
+    NewSeed { f: MNewSeed },
+    RecoverSeedName { f: MRecoverSeedName },
+    RecoverSeedPhrase { f: MRecoverSeedPhrase },
+    DeriveKey { f: MDeriveKey },
+    VVerifier { f: MVerifierDetails },
+    ManageNetworks { f: MManageNetworks },
+    NNetworkDetails { f: MNetworkDetails },
+    SignSufficientCrypto { f: MSignSufficientCrypto },
+    SelectSeedForBackup { f: MSeeds },
+    Documents,
+    KeyDetailsMulti { f: MKeyDetailsMulti },
+}
+
+#[derive(Clone, PartialEq)]
+pub struct Identity {
+    pub seed_name: String,
+    pub address_key: String,
+    pub public_key: String,
+    pub identicon: String,
+    pub has_pwd: bool,
+    pub path: String,
+    pub is_multiselect: bool,
+    pub base58: String,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MKeysCard {
+    pub address_key: String,
+    pub base58: String,
+    pub identicon: String,
+    pub has_pwd: bool,
+    pub path: String,
+    pub swiped: bool,
+    pub multiselect: bool,
+}
+
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct MNetworkCard {
+    pub title: String,
+    pub logo: String,
+}
+
+// TODO: This has to have a custom default.
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct MSeedKeyCard {
+    pub seed_name: String,
+    pub identicon: String,
+    pub address_key: String,
+    pub base58: String,
+    pub swiped: bool,
+    pub multiselect: bool,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MKeys {
+    pub set: Vec<MKeysCard>,
+    pub root: MSeedKeyCard,
+    pub network: MNetworkCard,
+    pub multiselect_mode: bool,
+    pub multiselect_count: String,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MSettings {
+    pub public_key: Option<String>,
+    pub identicon: Option<String>,
+    pub encryption: Option<String>,
+    pub error: Option<String>,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct History {
+    pub order: u32,
+    pub timestamp: String,
+    pub events: Vec<Event>,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MLog {
+    pub log: Vec<History>,
+    pub total_entries: u32,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MLogDetails {
+    pub timestamp: String,
+    pub events: Vec<Event>,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum TransactionType {
+    Sign,
+    Stub,
+    Read,
+    ImportDerivations,
+    Done,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct TransactionNetworkInfo {
+    pub network_title: String,
+    pub network_logo: String,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct TransactionAuthor {
```

Comment:
should have `has_password` field and be generally convertable to Address type (so they both can be visualized with the same UI component without extra effort)